library/java/io/envoyproxy/envoymobile/engine/JniLibrary.java

```diff
@@ -378,7 +378,23 @@ public static native Object callCertificateVerificationFromNative(byte[][] certC
   /**
    * Mimic a call to AndroidNetworkLibrary#clearTestRootCertificate from native code.
    * To be used for testing only.
-   *
    */
   public static native void callClearTestRootCertificateFromNative();
+
+  /**
+   * Mimic a call to AndroidCertVerifyResult#getStatus from native code.
+   * To be used for testing only.
+   *
+   * @param result AndroidCertVerifyResult to call getStatus onto.
+   */
+  public static native int callAndroidCertVerifyResultGetStatusFromNative(Object result);
```

Comment:
If this is supposed to be called from within tests only can we make it a part of a separate JNI interface - something such as `JNITestLibrary` or similar and add these 2 methods in there (maybe we could make JNITestLibrary to inherit from `JNILibrary`)? We could then include `JNILibrary` and these 2 methods in a test target only. That would allow us to avoid leaking testing utility methods in production code (and prevent accidental prod use of these methods).

Reply:
I've been trying to split the JNI library, but I've been hitting a bunch of Java runtime errors (I'm not entirely sure if I'm supposed to register the additional testing library somewhere?).

How strongly do you feel about this? As in, I completely agree that it's a good thing to do, but it's also relatively obvious that these are methods to be used only for testing. So, I'm not sure how much time is worth spending into this to get it "right/bullet proof".